src/addagent/validate.c

```diff
@@ -522,6 +526,37 @@ double OS_AgentAntiquity(const char *name, const char *ip){
     return output == OS_INVALID ? OS_INVALID : difftime(time(NULL), output);
 }
 
+/**
+ * @brief Returns the number of seconds since agent registration
+ *
+ * @param id The ID of the agent to get the registration time
+ * @retval On success, it returns the difference between the current time and the date_add field in the DB
+ * @retval OS_INVALID On error: failed to get the agent's info
+ * @retval 0 On error: failed to get the date_add field from JSON response
+ */
+time_t get_agent_registration_time(int id) {
+    cJSON *root = NULL;
+    cJSON *j_date_add = NULL;
+    time_t date_add = 0;
+
+    root = wdb_get_agent_info(id, NULL);
+
+    if (!root){
+        mdebug1("Failed to get agent info for agent '%d'", id);
+        return OS_INVALID;
+    }
+
+    j_date_add = cJSON_GetObjectItem(root->child, "date_add");
+    if(!j_date_add){
+        mdebug1("Failed to get registration time for agent '%d'.", id);
+    } else {
+        date_add = j_date_add->valueint;
+    }
+
+    cJSON_Delete(root);
+    return date_add <= 0 ? 0 : difftime(time(NULL), date_add);
```

Comment:
If we are going to return the delta between the actual time and the registration time we should rename this method to **get_agent_registered_time()**
In my opinion, it will be simpler and more flexible to have a method that only returns the registration time, and calculate the difference apart.

Reply:
The name has been changed, and the missing comment in the structure has been added.